### PR TITLE
integration: fix race in setting shared proxies

### PR DESCRIPTION
**Pull request by gyuho**
```
=== RUN   TestElectionWait
==================
WARNING: DATA RACE
Read at 0x00c420224f00 by goroutine 590:
  runtime.mapaccess2_fast64()
      /usr/local/go/src/runtime/hashmap_fast.go:147 +0x0
  github.com/coreos/etcd/integration.newClientV3()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_proxy.go:46 +0x146
  github.com/coreos/etcd/integration.NewClientV3()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster.go:552 +0x23d
  github.com/coreos/etcd/integration.makeClients.func1()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_lock_test.go:181 +0x8e
  github.com/coreos/etcd/integration.TestElectionWait.func1()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_election_test.go:46 +0x6c

Previous write at 0x00c420224f00 by goroutine 664:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:442 +0x0
  github.com/coreos/etcd/integration.newClientV3()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_proxy.go:46 +0x22d
  github.com/coreos/etcd/integration.NewClientV3()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster.go:552 +0x23d
  github.com/coreos/etcd/integration.makeClients.func1()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_lock_test.go:181 +0x8e
  github.com/coreos/etcd/integration.TestElectionWait.func1()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_election_test.go:46 +0x6c

Goroutine 590 (running) created at:
  github.com/coreos/etcd/integration.TestElectionWait()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_election_test.go:64 +0x42f
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 664 (running) created at:
  github.com/coreos/etcd/integration.TestElectionWait()
      /home/gyuho/go/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/v3_election_test.go:64 +0x42f
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
==================
```

https://github.com/coreos/etcd/blob/master/integration/v3_election_test.go#L44-L46 is causing races in `PASSES=grpcproxy ./test`

### CODE REVIEW

File: integration/cluster_proxy.go [hunk 1]
```diff
@@ -43,7 +52,9 @@ func newClientV3(cfg clientv3.Config) (*clientv3.Client, error) {
 		return nil, err
 	}
 
+	pmu.Lock()
 	proxies[c] = toGRPC(c)
```
Comment (heyitsanthony) [inline integration/cluster_proxy.go]: this assignment should probably happen in `toGRPC`. I think this deadlocks right now?

Comment (gyuho) [inline integration/cluster_proxy.go]: Right. Will fix! Thanks!

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**